spyder_memory_profiler/widgets/memoryprofiler.py

```diff
@@ -196,15 +196,21 @@ def select_file(self):
         self.redirect_stdio.emit(False)
         if filename:
             self.analyze(filename)
+
     def show_log(self):
         if self.output:
-            TextEditor(self.output, title=_("Memory profiler output"),
-                       readonly=True, size=(700, 500)).exec_()
+            editor = TextEditor(self.output, title=_("Memory profiler output"),
+                                readonly=True)
+            editor.show()
+            editor.exec_()
 
     def show_errorlog(self):
         if self.error_output:
-            TextEditor(self.error_output, title=_("Memory profiler output"),
-                       readonly=True, size=(700, 500)).exec_()
+            editor = TextEditor(self.error_output,
+                                title=_("Memory profiler output"),
+                                readonly=True)
+            editor.show()
+            editor.exec_()
```

Comment:
Why not use just `editor.exec_()` I beleived that called show first?

Reply:
Having just `editor.exec_()` does not work. I don't know exactly why, but it looks like that does not call `.show()`.

Reply:
Hmmm weird. Maybe adding a comment could help :-p ?

Reply:
My explanation was incomplete. After spyder-ide/spyder#10976, `TextEditor` inherits from `BaseDialog` which overrides `QDialog`'s `.show()` to dynamically resize the dialog window based on the screen size. However, `.exec()` does not call the override and thus, while it does show the dialog window, it does not dynamically resize it.

Anyway, I added a comment referencing the issue that kind of explains this.